### ISSUE TITLE
[State Sync] Add hardening features to public network bitswap setups

### ISSUE THREAD

**Issue by peterargue** [onflow/flow-go#5187]
## Problem Definition

TBD

## Proposed Solution

TBD

**Comment by bluesign** [comment]
Does this also allow p2p sharing? Observers in between sharing blobs I mean. If so there may need to add some additional protections.

**Comment by peterargue** [comment]
> Does this also allow p2p sharing? Observers in between sharing blobs I mean. If so there may need to add some additional protections.

yes, this would support p2p sharing of execution data between observers and ANs on the public network. What protections did you have in mind?

**Comment by bluesign** [comment]
I think reputation management for sender can be added, to verify they send correct data, with reasonable speed. ( unless this will be p2p for my own cluster )

**Comment by peterargue** [comment]
yes, absolutely. bitswap has some notion of reputation built in, and correctness of the data is validated after download. We will need to closely review the limits of the existing protections and fill any gaps.
